### ISSUE TITLE
[GlobalStep] "Product Catalog" fails to display limit of 6 for "Products per row and Rows per page" field as displayed on Page Editor

### ISSUE THREAD

**Issue by opr** [woocommerce/woocommerce#29347]
@gglobalstep commented on [Wed Mar 10 2021](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/3948)

## Bug Description:
"Product Catalog" fails to display limit of 6 for "Products per row and Rows per page" field as displayed on Page Editor

## Environment:
Woocommerce Version: WooCommerce 5.1 and 5.0
WordPress version: (v5.7)

PC:
Windows 10, Mac 10.14.6
Chrome(Version 88.0.4324.104) 
Firefox(Version 86.0)
Safari: v13.1.2(13609.3.5.1.5)

# Steps To Reproduce:
1. Create any test site using JN site.
1. Install and activate all the required plugins
1. Complete the onboarding setup wizard.
1. Add products/ import products from "Add my products" option on Woocommerce->Home Page
1. Go to Appearance->Customize.
1. Select "WooCommerce->Product Catalog" option.
1. Set the Products per rows to 7 or above and Rows per page to 7 or above.
1. Observe that "Product Catalog" fails to display limit of 6 for "Products per row" field as displayed on Page Editor

**Note**:
Observed for "Twenty-Twenty theme" for both the field mentioned above and for Storefront this issue is observed only for "Rows per page" field

## Actual Result:
"Product Catalog" fails to display limit of 6 for "Products per row and Rows per page" field as displayed on Page Editor

## Expected Result:
"Product Catalog" should display limit of 6 for "Products per row and Rows per page" field as displayed on Page Editor

## Video:

https://user-images.githubusercontent.com/41110392/110654251-fc934000-81e3-11eb-8be5-2b030356bdf8.mp4

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above. 

---

@opr commented on [Thu Mar 11 2021](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/3948#issuecomment-796890538)

Thanks for reporting this, I was able to reproduce, but I believe this to be an issue in WooCommerce Core, so I will move this issue there.

**Comment by wpericam** [comment]
Same issue in 29168909-hc

**Comment by wpericam** [comment]
WORKAROUND: 

In the instance above, we were able to work around this by changing the blog post settings here: 

<img width="928" alt="Screen Shot 2021-04-23 at 20 32 24" src="https://user-images.githubusercontent.com/44411833/115941660-4bcfc000-a474-11eb-95c4-0c1259561059.png">

**Comment by AashikP** [comment]
Hello @opr 

Thank you for sharing this report with us.

I took a look at my own test site, and I could replicate the same behavior with Twenty Twenty theme. 

However, this cannot be replicated with [Storefront Theme](https://woocommerce.com/storefront/?aff=10486&cid=1131038): https://d.pr/i/7Y9v2G

WooCommerce lets the theme decide the minimum and maximum number of rows to be allowed: [class-wc-shop-customizer.php#L484](https://github.com/woocommerce/woocommerce/blob/trunk/includes/customizer/class-wc-shop-customizer.php#L484)

The best way forward would be to configure it like Storefront does [class-storefront-woocommerce.php#L43](https://github.com/woocommerce/storefront/blob/trunk/inc/woocommerce/class-storefront-woocommerce.php#L43).

Perhaps we can report it here: https://core.trac.wordpress.org/search?q=twenty+twenty%3A

@wpericam I believe the issue in 29168909-hc was not related to this, as their report was that they cannot increase the limit, which was because of the blog post settings. This one is specifically about the display limit of 6, which WooCommerce does not force on the theme, but lets the theme decide the limit.

I'm closing this issue for now, but please feel free to comment on it in case I missed something. We’d be happy to take another look.

**Comment by rrennick** [comment]
@AashikP I'm re-opening this issue. WooCommerce core maintains the WC theme support for the default themes vs the theme developers at WordPress.org. For this issue - https://github.com/woocommerce/woocommerce/blob/trunk/includes/theme-support/class-wc-twenty-twenty.php

Adding `needs developer feedback`

**Comment by AashikP** [comment]
Thank you for clarifying, and re-opening the report, @rrennick  👍

**Comment by roykho** [comment]
Yes we can confirm that some of the default themes don't have specific CSS to handle this. Since we have compatibility logic to support all the different default themes, we should also fix this for compatibility reasons.

**Comment by ajbanx** [comment]
This is the worst part about Storefront. It's ridiculous we can only show 1 row of products and no one seems to know how to change that.
